includes/discovery/route.inc.php

```diff
@@ -14,154 +14,327 @@
  * You should have received a copy of the GNU General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>. */
 
+//We can use RFC1213 or IP-FORWARD-MIB or MPLS-L3VPN-STD-MIB
 
-//This file is a litle diferent, because the route depend of the vrf, not of the context, 
-//like the others, so if i use the context, you will have the same information n time the context how have the same VRF
-global $debug;
+use App\Models\Device;
+use LibreNMS\Util\IPv4;
 
+$ipForwardMibRoutesNumber = snmp_get($device, 'IP-FORWARD-MIB::inetCidrRouteNumber.0', '-Osqn');
 
-$ids = array();
+$ipForwardNb = snmp_get_multi($device, ['inetCidrRouteNumber.0', 'ipCidrRouteNumber.0'], '-OQUs', 'IP-FORWARD-MIB');
 
-// For the moment only will be cisco and the version 3
-if ($device['os_group'] == "cisco") {
-    //RFC1213-MIB
-    $mib = "RFC1213-MIB";
-    //IpRouteEntry
-    $vrfs_lite_cisco = array();
-
-    if (key_exists('vrf_lite_cisco', $device) && (count($device['vrf_lite_cisco']) != 0)) {
-        //i will only get one context of vrf, read the begin of this file
-        foreach ($device['vrf_lite_cisco'] as $vrf_lite) {
-            if (!key_exists($vrf_lite['vrf_name'], $vrfs_lite_cisco)) {
-                $vrfs_lite_cisco[$vrf_lite['vrf_name']] = $vrf_lite;
-            }
-        }
+//Get the configured max routes number
+$max_routes = 10000;
+if (null != (Config::get('routes.max_number'))) {
```

Comment:
This is the really long way of writing:
```php
$max_routes = Config::get('routes.max_number', 10000);
```

Reply:
default should probably match that in defaults.inc.php

Reply:
done